pages/basePage.js

```diff
@@ -0,0 +1,20 @@
+const { expect } = require("@playwright/test");
+
+exports.BasePage = class BasePage {
+  constructor(page) {
+    this.page = page;
+    this.shoppingCart = this.page.locator('[data-test="shopping-cart-link"]');
```

Comment:
You can define custom test id attribute in Playwright config, set [test-options-test-id-attribute](https://playwright.dev/docs/api/class-testoptions#test-options-test-id-attribute) and then just use [getByTestId](https://playwright.dev/docs/api/class-page#page-get-by-test-id).